### ISSUE TITLE
Reduced reasoning capability after #135621 (canonicalize_bool_expr)

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#136032]
### 🐛 Describe the bug

@isuruf, I'm trying to get you a repro, but I'm posting this in case it's obvious. Error looks like this.

![image](https://github.com/user-attachments/assets/d4fd305f-e205-419b-81fa-7169be736a9a)

This error is very strange because when I look at logs I actually do see 2*u48>0 already asserted. So I don't understand why we're failing to match lol.

The reporter says when https://github.com/pytorch/pytorch/issues/135621 is reverted they are able to get to the end

Internal xref: https://fb.workplace.com/groups/6829516587176185/permalink/7886514054809761/

### Versions

main

cc @chauhang @penguinwu

**Comment by isuruf** [comment]
Not obvious to me right now. Will take a look if you can provide a reproducer.

**Comment by ezyang** [comment]
So, this is not a repro, but here's the relevant deferred runtime asserts we have at the time we try to work out the symbol: [RuntimeAssert(expr=0 <= u48), RuntimeAssert(expr=0 < 1*u48)]

But this trivial looking repro doesn't work, smh.

```
import torch

torch._dynamo.config.capture_scalar_outputs = True

@torch.compile(fullgraph=True)
def f(x):
    y, y2 = x.tolist()
    torch._check(0 < 1 * y)
    if 2 * y > 0:
        return torch.tensor(True)
    else:
        return torch.tensor(False)

f(torch.tensor([23, 24]))
```

Weird...

**Comment by isuruf** [comment]
Ah, that's enough info. Will send a PR to fix it.
